File: ObjC_Example/Native/Views/NativeAdCellView.m
//
//  NativeAdCellView.m
//  ObjC_Example
//
//  Copyright (c) 2015年 FAN Communications. All rights reserved.
//

#import "NativeAdCellView.h"

@interface NativeAdCellView ()

@property (nonatomic, weak) IBOutlet UILabel *prTextLabel;
@property (nonatomic, weak) IBOutlet UILabel *longTextLabel;
@property (nonatomic, weak) IBOutlet UILabel *actionButtonTextLabel;
@property (nonatomic, weak) IBOutlet UIImageView *adImageView;

@end

@implementation NativeAdCellView

#pragma mark - NADNativeViewRendering.h

- (UILabel *)prTextLabel
{
    return _prTextLabel;
}

- (UILabel *)longTextLabel
{
    return _longTextLabel;
}

- (UILabel *)actionButtonTextLabel
{
    return _actionButtonTextLabel;
}

- (UIImageView *)adImageView
{
    return _adImageView;
}

@end

File: Sample/NendSDK_Swift_Sample/NativeAdCellView.swift
//
//  NativeAdCellView.swift
//  NendSDK_Sample
//
//  Copyright © 2015年 F@N Communications. All rights reserved.
//

import UIKit
import NendAd

class NativeAdCellView: UITableViewCell, NADNativeViewRendering {

    @IBOutlet fileprivate weak var nativeAdPrTextLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdLongTextLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdActionButtonTextLabel: UILabel!
    @IBOutlet fileprivate weak var nativeAdImageView: UIImageView!
    
    // MARK: - NADNativeViewRendering
    
    func prTextLabel() -> UILabel! {
        return self.nativeAdPrTextLabel
    }
    
    func longTextLabel() -> UILabel! {
        return self.nativeAdLongTextLabel
    }
    
    func actionButtonTextLabel() -> UILabel! {
        return self.nativeAdActionButtonTextLabel
    }
    
    func adImageView() -> UIImageView! {
        return self.nativeAdImageView
    }
}
